torchvision/prototype/datasets/_builtin/celeba.py

```diff
@@ -0,0 +1,200 @@
+            file_name="list_eval_partition.txt",
+        )
+        images = GDriveResource(
+            "0B7EVK8r0v71pZjFTYXZWM3FlRnM",
+            sha256="46fb89443c578308acf364d7d379fe1b9efb793042c0af734b6112e4fd3a8c74",
+            file_name="img_align_celeba.zip",
+        )
+        identities = GDriveResource(
+            "1_ee_0u7vcNLOfNLegJRHmolfH5ICW-XS",
+            sha256="c6143857c3e2630ac2da9f782e9c1232e5e59be993a9d44e8a7916c78a6158c0",
+            file_name="identity_CelebA.txt",
+        )
+        attributes = GDriveResource(
+            "0B7EVK8r0v71pblRyaVFSWGxPY0U",
+            sha256="f0e5da289d5ccf75ffe8811132694922b60f2af59256ed362afa03fefba324d0",
+            file_name="list_attr_celeba.txt",
+        )
+        bboxes = GDriveResource(
+            "0B7EVK8r0v71pbThiMVRxWXZ4dU0",
+            sha256="7487a82e57c4bb956c5445ae2df4a91ffa717e903c5fa22874ede0820c8ec41b",
+            file_name="list_bbox_celeba.txt",
+        )
+        landmarks = GDriveResource(
+            "0B7EVK8r0v71pd0FJY3Blby1HUTQ",
+            sha256="6c02a87569907f6db2ba99019085697596730e8129f67a3d61659f198c48d43b",
+            file_name="list_landmarks_align_celeba.txt",
+        )
+        return [splits, images, identities, attributes, bboxes, landmarks]
+
+    _SPLIT_ID_TO_NAME = {
+        "0": "train",
+        "1": "valid",
+        "2": "test",
+    }
+
+    def _filter_split(self, data: Tuple[str, str], *, split):
+        _, split_id = data
+        return self._SPLIT_ID_TO_NAME[split_id[0]] == split
+
+    def _csv_key(self, data: Tuple[str, Any]) -> str:
+        return data[0]
+
+    def _image_key(self, data: Tuple[str, Any]) -> str:
+        return pathlib.Path(data[0]).name
+
+    def _name_ann(self, data: Tuple[str, io.IOBase], *, name: str) -> Tuple[str, Dict[str, io.IOBase]]:
+        return data[0], {name: data[1]}
+
+    def _collate_partial_anns(self, data):
+        (id, anns), (_, partial_anns) = data
+        anns.update(partial_anns)
+        return id, anns
+
+    def _zip_anns_dp(self, anns_dps: List[IterDataPipe]) -> IterDataPipe:
+        anns_dps = {
+            name: CelebACSVParser(dp, has_header=has_header)
+            for (name, has_header), dp in zip(
+                (
+                    ("identity", False),
+                    ("attributes", True),
+                    ("bboxes", True),
+                    ("landmarks", True),
+                ),
+                anns_dps,
+            )
+        }
+        anns_dp: IterDataPipe
+        partial_anns_dps: Sequence[IterDataPipe]
+        anns_dp, *partial_anns_dps = [
+            Mapper(dp, self._name_ann, fn_kwargs=dict(name=name)) for name, dp in anns_dps.items()
+        ]
+        partial_anns_dp: IterDataPipe
+        for partial_anns_dp in partial_anns_dps:
+            anns_dp = KeyZipper(anns_dp, partial_anns_dp, lambda data: data[0], buffer_size=INFINITE_BUFFER_SIZE)
+            anns_dp = Mapper(anns_dp, self._collate_partial_anns)
```

Comment:
I will add an issue to data repo about the KeyZipper. This is a super anti-pattern design for stacked KeyZipper.

Reply:
Let me see if the files are perfectly aligned. If yes, we could use a `Zipper` instead, which is able to handle more than 2 datapipes at once.

Reply:
They are perfectly aligned :tada: The latest commits only need two `KeyZippers` to zip the splits, images, and annotations. We can't get around them at the moment.